### DISCUSSION TITLE
Flutter firebase auth with Microsoft in Mobile platforms

### DISCUSSION THREAD

**Discussion by rulila52** [firebase/flutterfire#7056] [Q&A] [answered]
Hi. I am trying to do auth with firebase microsoft on my android/ios app. I'm trying to call this functions:

```
UserCredential user = await auth.signInWithPopup(provider);


await auth.signInWithRedirect(provider);
UserCredential user = await auth.getRedirectResult();
```

But I get the following exception:
```
nhandled Exception: UnimplementedError: signInWithPopup() is only supported on web based platforms
```

Is any methods to do auth with firebase microsoft on mobile platforms?

**Comment by LeonanFragaLeonardo** [comment, marked as answer]
There is an another package:
https://pub.dev/packages/flutter_aad_oauth

Another way is to create auth with API:
https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow

If you have another option, please, share with us.

**Reply by rulila52** [reply]
Additional to your answer:

https://pub.dev/packages/firebase_auth_oauth

finally i use this package, but maybe your variants right too
